coding-the-legislation/35_periods.md

```diff
@@ -1,25 +1,95 @@
 # Periods
 
+A period can be a month, a year, `n` successive months or `n` successive years.
+
+
+## Periods for variable
+
 Most of the quantities calculated in openfisca can change over time. Therefore, each formula calculates a variable for a person (or a family, etc.) **for a given period**.
 
 This period is always the second argument of the formulas :
+
 ```py
+class var(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"some variable"
+    definition_period = MONTH
+
     def function(person, period):
         ...
 ```
 
-A period can be a month, a year, `n` successive months or `n` successive years.
+The size of the period is constrained by the class attribute `definition_period` :
+  - `definition_period = MONTH` : The variable may have a different value each month. For example, the salary of a person. The parameter `period` is guaranteed to be a whole month.
+  - `definition_period = YEAR` : The variable is yearly or has always the same value every month. For example, The input of a yearly declaration. The parameter `period` is guaranteed to be a whole year (from january 1st to 31th december).
+  - `definition_period = ETERNITY` : The value of the variable is constant. For example, the date of birth of a person never changes. There is no guarantee about `period` which must not be used.
+
+
+## Calculating dependencies for a period different than the one they are defined for
+
+Calling a formula with a period that is incompatible with the attribute `definition_period` will cause an error. For instance, if we assume that a person `salary` is paid monthly:
+
+```py
+class var(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"some yearly variable"
+    definition_period = YEAR
+
+    def function(person, period):
+        salary_past_year = person('salary', period) # THIS WILL BREAK !
+        ...
+```
+
+However, sometimes, we do need to estimate a variable for a different period that the one it is defined for.
+
+We may for example want to get the sum of the salaries perceived on the past year, or the past 3 months. The `ADD` option allows you to do it:
+
+```py
+class var(Variable):
+    column = FloatCol
+    entity = Person
+    label = u"some yearly variable"
+    definition_period = YEAR
+
+    def function(person, period):  # period is a year because definition_period = YEAR
```

Comment:
👍 this kind of “talk-through” comment is great in documentation, feel free to add some more!  😃